packages/core/src/state-management/state-manager.ts

```diff
@@ -95,14 +97,17 @@ export class StateManager {
     if (publish) {
       this.publishTip(state$);
     }
-    const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id);
     if (sync) {
-      const tip = await tip$.pipe(timeoutWith(3000, of(undefined))).toPromise();
+      const tip$ = this.dispatcher.messageBus.queryNetwork(state$.id);
+
+      const syncTimeout = (opts as LoadOpts).syncTimeoutMillis != undefined ? (opts as LoadOpts).syncTimeoutMillis : DEFAULT_SYNC_TIMEOUT
+      if (syncTimeout == 0) { // timeout of zero means don't wait for the response at all
```

Comment:
The rxjs docs don't seem to specify how `timeoutWith` handles a timeout of `0`, so I added this just to be safe, and to make the behavior explicit

Reply:
It handles it fine.